### PR TITLE
Spotahome Redis operator - fixes

### PR DESCRIPTION
**Pull request by ganochenkodg**
Resolving guide comments after tech-review

### CODE REVIEW

**Review by bourgeoisor — APPROVED** [review]

**Comment by bourgeoisor** [comment]
@ganochenkodg lemme know when this is good to merge in

File: databases/redis-spotahome/terraform/gke-autopilot/versions.tf
Comment (bourgeoisor) [inline databases/redis-spotahome/terraform/gke-autopilot/versions.tf:5]: Note that Renovate will try to automatically upgrade these again as soon as this PR merges in.

Is there any reason latest TF provider is not compatible?

If yes and we can't go around it, you might have to either a) update the CI job such that it fails if a new PR upgrading to 5 occurs again or b) tweak the Renovate config to permanently set this version to 4

**Review by bourgeoisor — CHANGES_REQUESTED** [review]

File: databases/redis-spotahome/terraform/gke-autopilot/versions.tf
Comment (ganochenkodg) [reply databases/redis-spotahome/terraform/gke-autopilot/versions.tf:5]: Yep, when I use ~> 5.0 i get failed:
Screenshot 2023-10-13 at 00 51 59

Other people from the team also got the same issue, so i decided to downgrade the version (it used to work with ~> 4.0)

Comment (bourgeoisor) [reply databases/redis-spotahome/terraform/gke-autopilot/versions.tf:5]: Would you mind updating the databases/ CI with an additional job for this "redis-spotahome" directory, with steps for each TF to do `terraform init` + `terraform validate`?

That way when Renovate creates another PR to bump to v5 of the provider, we'll immediately see those checks fail and won't merge it accidentally

Comment (ganochenkodg) [reply databases/redis-spotahome/terraform/gke-autopilot/versions.tf:5]: Added CI steps

**Review by bourgeoisor — APPROVED** [review]

**Pull request merged**
